app/Models/Rating.php

```diff
@@ -48,4 +62,92 @@ public function author()
     {
         return $this->belongsTo(\App\Models\User::class);
     }
+
+    /**
+     * Determines if the rating instance has a category given by the passed parameter
+     * @param $bitMask
+     * bitmask param should always be one of the bitmask constants defined on this model
+     * @return bool
+     */
+    public function hasCategory($bitMask)
+    {
+        return ($this->categories & $bitMask) === $bitMask;
+    }
+
+    /**
+     * Adds a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function addCategory($bitMask)
+    {
+        $this->categories |= $bitMask;
+    }
+
+    /**
+     * Removes a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function removeCategory($bitMask)
+    {
+        $this->categories &= ~$bitMask;
+    }
+
+    /**
+     * Returns category labels. Not recommended for large batches of requests, instead use hasCategory with bitmask
+     * @returns Collection of strings corresponding to the rating categories of this entity
+     */
+    public function getCategoryLabels()
+    {
+        $labels = collect();
+        $bitmasks = collect([
+            self::CREATIVE_BITMASK,
+            self::FLEXIBLE_BITMASK,
+            self::FRIENDLY_BITMASK,
+            self::HELPFUL_BITMASK,
+            self::PREPARED_BITMASK,
+        ]);
+
+        foreach ($bitmasks as $mask) {
+            if ($this->hasCategory($mask)) {
+                $labels->push($this->getCategoryLabelFromMask($mask));
+            }
+        }
+
+        return $labels;
+    }
```

Comment:
lets turn this into an [accessor](https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor) instead :) Something like `getLabelsAttribute()`

We can also save on evaluating all that control flow in [getCategoryLabelFromMask()](https://github.com/DnD-Montreal/session-tome/blob/9b95bc73c4487d94896d27342a6d92c85921facd/app/Models/Rating.php#L124) if we make the collection look like this: 

```php
$bitmasks = collect([
   self::CREATIVE_BITMASK => self::CREATIVE_LABEL,
   self::FLEXIBLE_BITMASK => self::FLEXIBLE_LABEL,
   self::FRIENDLY_BITMASK => self::FRIENDLY_LABEL,
   self::HELPFUL_BITMASK => self::HELPFUL_LABEL,
   self::PREPARED_BITMASK => self::PREPARED_LABEL,
])
```
 
Decomposing the `$this->categories` into an array of its powers of two and then calling `$bitmasks->only(...)`

Decomposition will look something like this: 
```php
$bit = 1;
$select = []
while ($bit <= $this->categories) {
   if ($bit & $this->categories) {
      $select[] = $bit
   }
   $bit = $bit << 1
}
```

Reply:
done! Its now an accessor so the labels can be found via `$rating->labels` which returns a collection of `bitmask => label`